### ISSUE TITLE
Maven BOM with profile fails to find `java.version`

### ISSUE THREAD

**Issue by stealthcode** [nebula-plugins/nebula-dependency-recommender-plugin#45]
I have pushed a code commit to [my fork](https://github.com/stealthcode/nebula-dependency-recommender-plugin/commit/47bd61d98cbfc332a98ca947406da9a47e829bc6) to reproduce the issue I see when trying to use my BOM. The error stops all recommending and causes failures in resolving dependencies (as there is no version specified. 

Error message:

```
org.apache.maven.model.building.ModelBuildingException: 1 problem was encountered while building the effective model for sample:recommender:1.0
[ERROR] Failed to determine Java version for profile java7 @ 


	at org.apache.maven.model.building.DefaultModelProblemCollector.newModelBuildingException(DefaultModelProblemCollector.java:195)
	at org.apache.maven.model.building.DefaultModelBuilder.build(DefaultModelBuilder.java:419)
	at org.apache.maven.model.building.DefaultModelBuilder.build(DefaultModelBuilder.java:371)
	at org.apache.maven.model.building.DefaultModelBuilder.build(DefaultModelBuilder.java:362)
	at org.apache.maven.model.building.DefaultModelBuilder.build(DefaultModelBuilder.java:232)
	at netflix.nebula.dependency.recommender.provider.MavenBomRecommendationProvider.getVersion(MavenBomRecommendationProvider.java:107)
	at netflix.nebula.dependency.recommender.provider.MavenBomRecommendationProviderSpec.recommendations are loaded from the dependencyManagement section of a BOM(MavenBomRecommendationProviderSpec.groovy:69)
```

The Apache maven model builder, when a profile is found that has an `<activation>` tag with a `<jdk>`nested within will fail because it looks for the `java.version` in the model builder request's system properties. This is not specified so the model builder throws an exception. 

I would have opened a pull request but I'm not sure where the java version should come from in the context of the call site. Any advice on how to handle this?

**Comment by stealthcode** [comment]
Should I just append the `java.version` to the context from the system properties and open a PR?

**Comment by rspieldenner** [comment]
PR would be great i'm not sure the best way to get the properties into the maven resolver

**Comment by rspieldenner** [comment]
might be as easy as adding `request.setSystemProperties(System.properties)`

**Comment by kgeis** [comment]
I am having this problem as well. Apparently it was a problem in Grails 2.x. The solution in that case was as @rspieldenner says.

Documented here:

[Grails, Maven and JDK-based Profile Activation](http://jdpgrailsdev.github.io/blog/2014/05/27/grails_maven_jdk_profile.html)

[grails-core: JDK-based Maven profile activiation fix #492](https://github.com/grails/grails-core/pull/492)
